CDS/js/youtube/youtube-upload-lib.js

```diff
@@ -0,0 +1,114 @@
+var Promise = require('promise');
+var fs = require('fs');
+var youtubeAuth = require('./youtube-auth.js');
+var googleapis = require('googleapis');
+var OAuth2 = googleapis.auth.OAuth2;
+var dataStore = require('../Datastore');
+
+const YOUTUBE_API_VERSION = 'v3';
+
+function getMetadata(connection) {
+
+    return new Promise((fulfill, reject) => {
+
+        if (!process.env.title) {
+            reject(new Error('Cannot upload to youtube: missing a title'));
+        }
+
+        if (!process.env.description) {
+            reject(new Error('Cannot upload to youtube: missing a description'));
+        }
+
+        if (!process.env.category_id) {
+            reject(new Error('Cannot upload to youtube: missing a category id'));
+        }
+
+        fulfill(dataStore.substituteStrings(connection, [process.env.title, process.env.description, process.env.category_id, process.env.access])
+        .then((substitutedStrings) => {
+            var title, description, category_id, status;
+            [title, description, category_id, status] = substitutedStrings;
+
+            return {
+                snippet: {
+                    title: title,
+                    description: description,
+                    categoryId: category_id
+                },
+                status: { privacyStatus: status ? status : 'private'}
+            }
+        }));
+    });
+}
+
+function getYoutubeData(connection) {
+    return this.getMetadata(connection)
+    .then((metadata) => {
+        const mediaPath = process.env.cnf_media_file;
+
+        if (!process.env.cnf_media_file) {
+            throw new Error('Cannot upload to youtube: missing media file path');
```

Comment:
line 48 should also be process.env.cf_media_file.  I'd be a little more explicit in the error message, and warn the user that they should check that <take-files>media</take-files> has been specified in the route.